playbooks/common/openshift-cluster/openshift_hosted.yml

```diff
@@ -1,5 +1,7 @@
 - name: Create Hosted Resources
   hosts: oo_first_master
+  pre_tasks:
+  - set_fact:
+      openshift_hosted_router_registryurl: "{{ hostvars[groups.oo_first_master.0].openshift.master.registry_url }}"
```

Comment:
I suspect this is similar to the issue when things are being set in vars...  The templates are being evaluated when they are used rather than when they are defined (as in the case of set_fact).